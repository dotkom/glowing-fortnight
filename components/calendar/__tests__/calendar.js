import React from 'react';
import Calendar from '../calendar';
import calendarData from './calendarData.json';
import { shallow, mount } from 'enzyme';

const beforeDate = new Date('2017-08-01T00:00:00Z').valueOf();
const duringDate = new Date('2017-08-20T16:00:00Z').valueOf();
const afterDate = new Date('2017-09-01T00:00:00Z').valueOf();

it('renders correctly before fadderuker', () => {
  Date.now = jest.fn(() => beforeDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders correctly during fadderuker', () => {
  Date.now = jest.fn(() => duringDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders correctly after fadderuker', () => {
  Date.now = jest.fn(() => afterDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders past events after click', () => {
  Date.now = jest.fn(() => duringDate);
  const wrapper = shallow(<Calendar events={calendarData} error={null} />);
  wrapper.find('.cal-button--preDays').simulate('click');
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders error', () => {
  const wrapper = shallow(<Calendar events={[]} error="Error message" />);
  expect(wrapper.getElements()).toMatchSnapshot();
});

it('renders loading message', () => {
  const wrapper = shallow(<Calendar events={[]} error={null} />);
  expect(wrapper.getElements()).toMatchSnapshot();
});
