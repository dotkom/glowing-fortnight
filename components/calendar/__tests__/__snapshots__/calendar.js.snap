// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly after fadderuker 1`] = `
Array [
  <div>
    <div
      className="cal-timeline"
    />
    
    <button
      className="cal-button--preDays"
      onClick={[Function]}
    >
      Vis tidligere arrangementer
    </button>
  </div>,
]
`;

exports[`renders correctly before fadderuker 1`] = `
Array [
  <div>
    <div
      className="cal-timeline"
    />
    
    
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "Velkommen til Trondheim, informatikk og Online

Lurer du på hvordan du skal komme deg rundt på Gløshaugen? Hvor de viktigste stedene i Trondheim er? Er det første gang du flytter hjemme fra og du lurer på hvordan i all verden du skal klare deg på bare lånet? Hva er det med de mattefagene alle snakker om? Og trenger man egentlig kjøpe pensum-bøker, det er jo så dyrt! Da er ny-på-informatikk-kurset noe for deg! En kjapp gjennomgang av det viktigste du trenger å vite som ny student på informatikk, hvordan du gjør deg klar til studiestart, fadderukene og alle mulighetene Trondheim har å by på.

Sted: Kommer",
            "end_time": "2017-08-14T14:00:00Z",
            "id": 77,
            "index": 0,
            "start_time": "2017-08-14T13:00:00Z",
            "title": "Ny-på-informatikk",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "Nå er det på tide å finne frem sin indre skatteplyndrer, for i dag er det skattejakt! Møt opp til anvist tid utenfor P15.",
            "end_time": "2017-08-20T15:00:00Z",
            "id": 76,
            "index": 1,
            "start_time": "2017-08-20T12:00:00Z",
            "title": "Skattejakt",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "Blåmandag er et kjent fenomen for de fleste. Denne mandagen blir alt annet enn blå, for i dag er det duket for Hybel-Hybel!",
            "end_time": "2017-08-21T21:00:00Z",
            "id": 74,
            "index": 2,
            "start_time": "2017-08-21T15:00:00Z",
            "title": "Hybel til Hybel",
          },
          Object {
            "content": "Etter konkurranserunden, er det etterfest på Cafè 3b.",
            "end_time": "2017-08-22T01:59:00Z",
            "id": 75,
            "index": 3,
            "start_time": "2017-08-21T21:00:00Z",
            "title": "Etterfest",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag er det duket for den legendariske Facebook-Festen! Faddergruppene står for faddervors før alle møtes på Nattergalen.",
            "end_time": "2017-08-22T21:59:00Z",
            "id": 72,
            "index": 4,
            "start_time": "2017-08-22T15:00:00Z",
            "title": "Facebook-Fest",
          },
          Object {
            "content": "Alternativet til kveldens fest er en tur på kino, etterfulgt av felles middag. Perfekt mulighet for å bli kjent med medstudentene dine.",
            "end_time": "2017-08-22T21:30:00Z",
            "id": 73,
            "index": 5,
            "start_time": "2017-08-22T16:00:00Z",
            "title": "Kino og Middag",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag tar vi det rolig, og griller i høyskoleparken. Gratis mat = Studentens beste venn.",
            "end_time": "2017-08-23T15:00:00Z",
            "id": 69,
            "index": 6,
            "start_time": "2017-08-23T12:00:00Z",
            "title": "Grilling",
          },
        ]
      }
      highlightIndex={0}
    />
  </div>,
]
`;

exports[`renders correctly during fadderuker 1`] = `
Array [
  <div>
    <div
      className="cal-timeline"
    />
    
    <button
      className="cal-button--preDays"
      onClick={[Function]}
    >
      Vis tidligere arrangementer
    </button>
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "Blåmandag er et kjent fenomen for de fleste. Denne mandagen blir alt annet enn blå, for i dag er det duket for Hybel-Hybel!",
            "end_time": "2017-08-21T21:00:00Z",
            "id": 74,
            "index": 0,
            "start_time": "2017-08-21T15:00:00Z",
            "title": "Hybel til Hybel",
          },
          Object {
            "content": "Etter konkurranserunden, er det etterfest på Cafè 3b.",
            "end_time": "2017-08-22T01:59:00Z",
            "id": 75,
            "index": 1,
            "start_time": "2017-08-21T21:00:00Z",
            "title": "Etterfest",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag er det duket for den legendariske Facebook-Festen! Faddergruppene står for faddervors før alle møtes på Nattergalen.",
            "end_time": "2017-08-22T21:59:00Z",
            "id": 72,
            "index": 2,
            "start_time": "2017-08-22T15:00:00Z",
            "title": "Facebook-Fest",
          },
          Object {
            "content": "Alternativet til kveldens fest er en tur på kino, etterfulgt av felles middag. Perfekt mulighet for å bli kjent med medstudentene dine.",
            "end_time": "2017-08-22T21:30:00Z",
            "id": 73,
            "index": 3,
            "start_time": "2017-08-22T16:00:00Z",
            "title": "Kino og Middag",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag tar vi det rolig, og griller i høyskoleparken. Gratis mat = Studentens beste venn.",
            "end_time": "2017-08-23T15:00:00Z",
            "id": 69,
            "index": 4,
            "start_time": "2017-08-23T12:00:00Z",
            "title": "Grilling",
          },
        ]
      }
      highlightIndex={0}
    />
  </div>,
]
`;

exports[`renders error 1`] = `
Array [
  <p
    className="component"
  >
    En uventet feil har oppstått ved henting av program. Vennligst prøv igjen senere.
  </p>,
]
`;

exports[`renders loading message 1`] = `
Array [
  <h2
    className="component"
  >
    Laster inn kalender
  </h2>,
]
`;

exports[`renders past events after click 1`] = `
Array [
  <div>
    <div
      className="cal-timeline"
    />
    <div
      className="cal-section--preDays"
    >
      <Day
        eventClickHandler={[Function]}
        events={
          Array [
            Object {
              "content": "Velkommen til Trondheim, informatikk og Online

Lurer du på hvordan du skal komme deg rundt på Gløshaugen? Hvor de viktigste stedene i Trondheim er? Er det første gang du flytter hjemme fra og du lurer på hvordan i all verden du skal klare deg på bare lånet? Hva er det med de mattefagene alle snakker om? Og trenger man egentlig kjøpe pensum-bøker, det er jo så dyrt! Da er ny-på-informatikk-kurset noe for deg! En kjapp gjennomgang av det viktigste du trenger å vite som ny student på informatikk, hvordan du gjør deg klar til studiestart, fadderukene og alle mulighetene Trondheim har å by på.

Sted: Kommer",
              "end_time": "2017-08-14T14:00:00Z",
              "id": 77,
              "index": 0,
              "start_time": "2017-08-14T13:00:00Z",
              "title": "Ny-på-informatikk",
            },
          ]
        }
        highlightIndex={0}
      />
      <Day
        eventClickHandler={[Function]}
        events={
          Array [
            Object {
              "content": "Nå er det på tide å finne frem sin indre skatteplyndrer, for i dag er det skattejakt! Møt opp til anvist tid utenfor P15.",
              "end_time": "2017-08-20T15:00:00Z",
              "id": 76,
              "index": 1,
              "start_time": "2017-08-20T12:00:00Z",
              "title": "Skattejakt",
            },
          ]
        }
        highlightIndex={0}
      />
    </div>
    <button
      className="cal-button--preDays"
      onClick={[Function]}
    >
      Skjul tidligere arrangementer
    </button>
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "Blåmandag er et kjent fenomen for de fleste. Denne mandagen blir alt annet enn blå, for i dag er det duket for Hybel-Hybel!",
            "end_time": "2017-08-21T21:00:00Z",
            "id": 74,
            "index": 0,
            "start_time": "2017-08-21T15:00:00Z",
            "title": "Hybel til Hybel",
          },
          Object {
            "content": "Etter konkurranserunden, er det etterfest på Cafè 3b.",
            "end_time": "2017-08-22T01:59:00Z",
            "id": 75,
            "index": 1,
            "start_time": "2017-08-21T21:00:00Z",
            "title": "Etterfest",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag er det duket for den legendariske Facebook-Festen! Faddergruppene står for faddervors før alle møtes på Nattergalen.",
            "end_time": "2017-08-22T21:59:00Z",
            "id": 72,
            "index": 2,
            "start_time": "2017-08-22T15:00:00Z",
            "title": "Facebook-Fest",
          },
          Object {
            "content": "Alternativet til kveldens fest er en tur på kino, etterfulgt av felles middag. Perfekt mulighet for å bli kjent med medstudentene dine.",
            "end_time": "2017-08-22T21:30:00Z",
            "id": 73,
            "index": 3,
            "start_time": "2017-08-22T16:00:00Z",
            "title": "Kino og Middag",
          },
        ]
      }
      highlightIndex={0}
    />
    <Day
      eventClickHandler={[Function]}
      events={
        Array [
          Object {
            "content": "I dag tar vi det rolig, og griller i høyskoleparken. Gratis mat = Studentens beste venn.",
            "end_time": "2017-08-23T15:00:00Z",
            "id": 69,
            "index": 4,
            "start_time": "2017-08-23T12:00:00Z",
            "title": "Grilling",
          },
        ]
      }
      highlightIndex={0}
    />
  </div>,
]
`;
